src/components/Booking/container.js

```diff
@@ -48,11 +55,28 @@ export default Container => class extends Component {
         startDate: holiday.startDate || chosenDate,
         endDate: holiday.endDate || chosenDate,
         halfDay: holiday.halfDay,
+        wasHalfDay: holiday.halfDay,
+        duration: holiday.duration,
       },
       booked,
     });
   }
 
+  componentWillUnmount() {
+    this.sub.remove();
+  }
+
+  getData = () => {
+    getUserEvents()
+      .then(events => this.setState({ events }));
+
+    getRemainingHolidays()
+      .then(remainingHolidays => this.setState({ remainingHolidays }));
+
+    userProfile()
+      .then(user => this.setState({ user, eventsLoaded: true }));
+  }
```

Comment:
This may cause issues as these inner functions won't be running asynchronously. 

Try making use of `Promise.all()` here
```
loadData = () => {
  Promise.all([
    getUserEvents, 
    getRemainingHolidays, 
    userProfile,
  ]).then((res) => {
    this.setState({
      events: res[0],
      remainingHolidays: res[1],
      user: res[2],
      eventsLoaded: true,
    })
  })
}
```

Just double check that works as expected lol

Reply:
thanks :) I have amended same